public/pages/EditFeatures/components/CustomAggregation/CustomAggregation.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import React from 'react';
+//@ts-ignore
+import { EuiFormRow, EuiCodeEditor } from '@elastic/eui';
+import { Field, FieldProps } from 'formik';
+import { isInvalid, getError } from '../../../../utils/utils';
+
+import 'brace/ext/language_tools';
+
+interface CustomAggregationProps {
+  index: number;
+}
+
+export const validateQuery = (value: string) => {
+  try {
+    JSON.parse(value);
+  } catch (err) {
+    console.log('Returning error', err);
+    return 'Invalid JSON';
+  }
+};
+
+export const CustomAggregation = (props: CustomAggregationProps) => {
+  return (
+    <Field
+      id={`featureList.${props.index}.aggregationQuery`}
+      name={`featureList.${props.index}.aggregationQuery`}
+      validate={validateQuery}
```

Comment:
what will happen if `Invalid JSON` is returned by `validateQuery`?

Reply:
We will show error below the code editor like "Invalid JSON" and will block saving and previewing .